openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -53,40 +54,54 @@
  * @since 2.1.0
  */
 public class OASModifier {
-    OpenAPI openapi;
     List<Diagnostic> diagnostics = new ArrayList<>();
 
-    public OASModifier(OpenAPI openapi) {
-        this.openapi = openapi;
-    }
 
     public List<Diagnostic> getDiagnostics() {
         return diagnostics;
     }
 
-    //modifyWithBallerinaConventions 
-    public OpenAPI modifyWithBallerinaConventions() {
+    public Optional<Map<String, String>> getProposedNameList(OpenAPI openapi) {
+        Map<String, String> nameMap = new HashMap<>();
+        if (openapi.getComponents() == null) {
+            return Optional.empty();
+        }
+        Components components = openapi.getComponents();
+        Map<String, Schema> schemas = components.getSchemas();
+        if (schemas == null) {
+            return Optional.empty();
+        }
+        Set<String> modifiedSchemaNames = new HashSet<>();
+
+        for (Map.Entry<String, Schema> schemaEntry: schemas.entrySet()) {
+            //modified logic should come to here
+            String schemaName = schemaEntry.getKey();
+            String modifiedName = getValidNameForType(schemaEntry.getKey());
+            if (!schemaName.equals(modifiedName) && (schemas.containsKey(modifiedName) ||
+                    modifiedSchemaNames.contains(modifiedName))) {
+                // todo: check the duplication, till providing the name this will give same name as it is.
+                continue;
+            }
+            modifiedSchemaNames.add(modifiedName);
+            nameMap.put(schemaEntry.getKey(), modifiedName);
+        }
+        return Optional.of(nameMap);
+    }
+
+    public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi, Map<String, String> nameMap) {
```

Comment:
This function is too long, try to split this appropriately

Reply:
Address via https://github.com/ballerina-platform/openapi-tools/pull/1762/commits/deca2c23959600fefc619afbde162eecec1036f0